calico/felix/felix.py

```diff
@@ -547,21 +547,27 @@ def _update_endpoint(self, endpoint, fields):
             for addr in fields['addrs']:
                 addresses.add(Address(addr))
         except KeyError:
-            log.error("Missing addrs or IP in addrs for endpoint %s, data %s",
-                      self.uuid, fields)
-            raise InvalidRequest("No valid address for endpoint %s")
+            log.error("Missing addr or IP in addrs for endpoint %s : %s",
+                      endpoint.uuid, fields)
+            raise InvalidRequest("No valid address for endpoint %s" %
+                                 endpoint.uuid)
 
         if mac is None:
-            log.error("No mac address for endpoint %s")
-            raise InvalidRequest("No mac address for endpoint %s")
+            log.error("No mac address for endpoint %s : %s",
+                      endpoint.uuid,
+                      fields)
+            raise InvalidRequest("No mac address for endpoint %s" %
+                                 endpoint.uuid)
 
         if state is None:
-            log.error("No state for endpoint %s")
-            raise InvalidRequest("No state for endpoint %s")
+            log.error("No state for endpoint %s : %s",
+                      endpoint.uuid,
+                      fields)
+            raise InvalidRequest("No state for endpoint %s" % endpoint.uuid)
 
         if state not in Endpoint.STATES:
-            log.error("Invalid state %s for endpoint %s : %s" %
-                      (state, endpoint.uuid, state))
+            log.error("Invalid state %s for endpoint %s : %s",
+                      state, endpoint.uuid, fields)
```

Comment:
Yes, I think so.

We are trying to report the invalid state, the endpoint it applied to and the entire request. Not guaranteeing that it's really not got some obvious dumb typo that I just cannot see, but it looks right.